src/VscPreviewExtension/Microsoft.DocAsCode.Dfm.VscPreview/DocfxProcessor.cs

```diff
@@ -13,69 +14,130 @@ namespace Microsoft.DocAsCode.Dfm.VscPreview
+            }
+
+            if (string.IsNullOrEmpty(previewFilePath))
+            {
+                throw new DocfxPreviewException("Preview file path should not be null or empty");
+            }
+            if (string.IsNullOrEmpty(pageUpdateJsFilePath))
+            {
+                throw new DocfxPreviewException("Page Update js file path should not be null or empty");
+            }
 
-            var markupResult = DfmMarkup(baseDir, relativePath, markdownContent.ToString());
+            PreviewJsonConfig config = PreviewCommand.ParsePreviewCommand(baseDir);
 
-            string originHtmlPath = FindOriginHtml(baseDir, relativePath, config.OutputFolder);
+            // Path of Html which generated from target markdown file
+            string targetHtmlPath = Path.Combine(Path.GetDirectoryName(previewFilePath),
+                Path.GetFileNameWithoutExtension(relativePath) + ".html");
 
-            if (string.IsNullOrEmpty(originHtmlPath))
+            if (string.IsNullOrEmpty(targetHtmlPath))
             {
                 // TODO: If the return value is not a complete Html, it should be contacted with an Html header and tail
                 return markupResult;
             }
 
-            string htmlString = File.ReadAllText(originHtmlPath);
+            string htmlString = File.ReadAllText(targetHtmlPath);
 
             CQ dom = htmlString;
 
-            // Update markup result
-            dom.Select(config.MarkupResultLocation).Html(markupResult);
+            CQ addElements = $"<script type='text/javascript' src='{pageUpdateJsFilePath}'></script>" +
+                             $"<meta name='pageRefreshFunctionName' content ='{config.PageRefreshFunctionName}'>" +
+                             $"<meta name='port' content='{config.Port}'>" +
+                             $"<meta name='filePath' content='{relativePath}'>" +
+                             $"<meta name='markupTagType' content='{config.MarkupTagType}'>" +
+                             $"<meta name='markupClassName' content='{config.MarkupClassName}'>";
 
+            foreach (var addElement in addElements)
+            {
+                dom.Select(addElement.NodeName)
+                    .Last().
+                    After(addElement);
+            }
+
+            // Replace 'https' to 'http' for that VS Code don't support reference which use https protocol now
+            // Replace reference relative path to absolute path
             foreach (var item in config.References)
             {
                 dom.Select(item.Key).Each((i, e) =>
                 {
                     var path = e.GetAttribute(item.Value);
-                    e.SetAttribute(item.Value, GetAbsolutePath(originHtmlPath, path));
+                    if (string.IsNullOrEmpty(path))
+                        return;
+                    if (path.StartsWith("https"))
+                    {
+                        e.SetAttribute(item.Value, path.Replace("https", "http"));
```

Comment:
this will replace all occurrences. Do you mean url will cover more than one "https"?

---
In reply to: [105577236](https://github.com/dotnet/docfx/pull/1397#discussion_r105577236) [](ancestors = 105577236)